dev/com.ibm.ws.ejbcontainer.security/src/com/ibm/ws/ejbcontainer/security/internal/EJBSecurityCollaboratorImpl.java

```diff
@@ -737,4 +758,31 @@ public void componentMetaDataCreated(MetaDataEvent<ComponentMetaData> event) {
     public void componentMetaDataDestroyed(MetaDataEvent<ComponentMetaData> event) {
 
     }
+
+    @FFDCIgnore(InterruptedException.class)
+    private void waitForSecurity() {
+        final boolean isTraceOn = TraceComponent.isAnyTracingEnabled();
+
+        if (waitedForSecurity || securityReadyService.isSecurityReady()) {
+            waitedForSecurity = true;
+            return;
+        }
+
+        CountDownLatch securityReadyCDL = securityReadyService.getSecurityReadyCDL();
```

Comment:
Instead of obtaining the CounDownLatch from the SecurityReadyService, it may be cleaner to just add an "await" or "wait" method to the SecurityReadyService.  With this change, callers would not need to care about the implementation detail of using a CountDownLatch.... and the below code would not need to be duplicated in all callers.

This would also avoid the possibility that a caller may be tempted to cache the CoundDownLatch, which would be incorrect since the SecurityReadyService may replace it.